lib/queues.py

```diff
@@ -153,52 +154,53 @@ def _on_message(self, data):
-
-            # Check if it's a valid platform
-            if not self.is_valid_platform(tree, update['platform']):
-                raise ValueError('Cancel update request due to invalid platform: {}'.
-                                 format(update['platform']))
-
-            # Check if it's a valid locale
-            if not self.is_valid_locale(tree, update['locale']):
-                raise ValueError('Cancel update request due to invalid locale: {}'.
-                                 format(update['locale']))
-
-            self.callback(allowed_testruns=['update'],
-                          platform=update['platform'],
-                          product=update['appName'].lower(),
-                          branch=update['branch'],
-                          locale=update['locale'],
-                          buildid=update['from_buildid'],
-                          revision=update['revision'],
-                          target_version=update['version'],
-                          target_buildid=update['to_buildid'],
-                          json_data=update,
-                          )
+            try:
+                # Check if its a valid tree
+                tree = update['branch']
+                if not self.is_valid_tree(tree):
+                    raise ValueError('Cancel update request due to invalid tree: {}'.
+                                     format(tree))
+
+                # Check if it's a valid product
+                if not self.is_valid_product(tree, update['appName'].lower()):
+                    raise ValueError('Cancel update request due to invalid product: {}'.
+                                     format(update['appName'].lower()))
+
+                # Check if it's a valid platform
+                if not self.is_valid_platform(tree, update['platform']):
+                    raise ValueError('Cancel update request due to invalid platform: {}'.
+                                     format(update['platform']))
+
+                # Check if it's a valid locale
+                if not self.is_valid_locale(tree, update['locale']):
+                    raise ValueError('Cancel update request due to invalid locale: {}'.
+                                     format(update['locale']))
+
+                self.callback(allowed_testruns=['update'],
+                              platform=update['platform'],
+                              product=update['appName'].lower(),
+                              branch=update['branch'],
+                              locale=update['locale'],
+                              buildid=update['from_buildid'],
+                              revision=update['revision'],
+                              target_version=update['version'],
+                              target_buildid=update['to_buildid'],
+                              json_data=update,
+                              )
+            except ValueError, e:
```

Comment:
exceptions should be written out as `ValueError as e`. This is valid in python 2 but not python 3.